test/unit/test_model_server.py

```diff
@@ -0,0 +1,330 @@
+    torchserve._create_torchserve_config_file()
+
+    write_file.assert_called_once_with(
+        torchserve.TS_CONFIG_FILE, generate_ts_config_props.return_value
+    )
+
+
+@patch("sagemaker_inference.utils.read_file", return_value=DEFAULT_CONFIGURATION)
+@patch("sagemaker_inference.environment.Environment")
+def test_generate_ts_config_properties(env, read_file):
+    model_server_timeout = "torchserve_timeout"
+    model_server_workers = "torchserve_workers"
+    http_port = "http_port"
+
+    env.return_value.model_server_timeout = model_server_timeout
+    env.return_value.model_sever_workerse = model_server_workers
+    env.return_value.inference_http_port = http_port
+
+    ts_config_properties = torchserve._generate_ts_config_properties()
+
+    inference_address = "inference_address=http://0.0.0.0:{}\n".format(http_port)
+    server_timeout = "default_response_timeout={}\n".format(model_server_timeout)
+
+    read_file.assert_called_once_with(torchserve.DEFAULT_TS_CONFIG_FILE)
+
+    assert ts_config_properties.startswith(DEFAULT_CONFIGURATION)
+    assert inference_address in ts_config_properties
+    assert server_timeout in ts_config_properties
+
+
+@patch("sagemaker_inference.utils.read_file", return_value=DEFAULT_CONFIGURATION)
+@patch("sagemaker_inference.environment.Environment")
+def test_generate_ts_config_properties_default_workers(env, read_file):
+    env.return_value.model_server_workers = None
+
+    ts_config_properties = torchserve._generate_ts_config_properties()
+
+    workers = "default_workers_per_model={}".format(None)
+
+    read_file.assert_called_once_with(torchserve.DEFAULT_TS_CONFIG_FILE)
+
+    assert ts_config_properties.startswith(DEFAULT_CONFIGURATION)
+    assert workers not in ts_config_properties
+
+
+@patch("sagemaker_inference.utils.read_file", return_value=DEFAULT_CONFIGURATION)
+@patch("sagemaker_inference.environment.Environment")
+def test_generate_ts_config_properties_multi_model(env, read_file):
+    env.return_value.model_server_workers = None
+
+    torchserve.ENABLE_MULTI_MODEL = True
+    ts_config_properties = torchserve._generate_ts_config_properties()
+    torchserve.ENABLE_MULTI_MODEL = False
+
+    workers = "default_workers_per_model={}".format(None)
+
+    read_file.assert_called_once_with(torchserve.MME_TS_CONFIG_FILE)
+
+    assert ts_config_properties.startswith(DEFAULT_CONFIGURATION)
+    assert workers not in ts_config_properties
+
+
+@patch("signal.signal")
+def test_add_sigterm_handler(signal_call):
+    ts = Mock()
+
+    torchserve._add_sigterm_handler(ts)
+
+    mock_calls = signal_call.mock_calls
+    first_argument = mock_calls[0][1][0]
+    second_argument = mock_calls[0][1][1]
+
+    assert len(mock_calls) == 1
+    assert first_argument == signal.SIGTERM
+    assert isinstance(second_argument, types.FunctionType)
+
+
+@patch("subprocess.check_call")
+def test_install_requirements(check_call):
+    torchserve._install_requirements()
```

Comment:
maybe assert that `check_call` was called with the expected args?

Reply:
Done. This depends in the python binary in test env. 

Have added an assertion for validating the pip install